solr/core/src/java/org/apache/solr/jersey/SolrRequestAuthorizer.java

```diff
@@ -66,7 +67,9 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
         (AuthorizationContext.RequestType)
             requestContext.getProperty(RequestContextKeys.REQUEST_TYPE);
     final List<String> collectionNames =
-        (List<String>) requestContext.getProperty(RequestContextKeys.COLLECTION_LIST);
+        Objects.requireNonNullElse(
```

Comment:
hmm. hit this during unit testing. looked again and maybe it's due to my test rather than the value being null. will check again and remove this if it's not needed.